File: internal/util/env.ts
import { createReadStream } from "node:fs";
import { access } from "node:fs/promises";
import readline from "node:readline";

export async function loadEnvFile(path: string): Promise<void> {
  if (!path) return;
  try {
    await access(path);
  } catch {
    return;
  }

  const stream = createReadStream(path, { encoding: "utf8" });
  const rl = readline.createInterface({ input: stream, crlfDelay: Infinity });

  for await (const raw of rl) {
    let line = raw.trim();
    if (!line || line.startsWith("#")) continue;
    if (line.startsWith("export ")) {
      line = line.slice("export ".length).trim();
    }
    const idx = line.indexOf("=");
    if (idx === -1) continue;
    const key = line.slice(0, idx).trim();
    if (!key || Object.prototype.hasOwnProperty.call(process.env, key)) continue;
    let val = line.slice(idx + 1).trim();
    if (val.length >= 2) {
      const first = val[0];
      const last = val[val.length - 1];
      if ((first === '"' && last === '"') || (first === "'" && last === "'")) {
        val = val.slice(1, -1);
      }
    }
    process.env[key] = val;
  }
}

File: internal/util/env.py
from __future__ import annotations

import os
from pathlib import Path


def load_env_file(path: str) -> None:
    if not path:
        return
    file_path = Path(path)
    if not file_path.exists():
        return
    try:
        with file_path.open("r", encoding="utf-8") as handle:
            for raw in handle:
                line = raw.strip()
                if not line or line.startswith("#"):
                    continue
                if line.startswith("export "):
                    line = line[len("export ") :].strip()
                if "=" not in line:
                    continue
                key, value = line.split("=", 1)
                key = key.strip()
                if not key or key in os.environ:
                    continue
                value = value.strip()
                if len(value) >= 2 and (
                    (value[0] == '"' and value[-1] == '"')
                    or (value[0] == "'" and value[-1] == "'")
                ):
                    value = value[1:-1]
                os.environ[key] = value
    except FileNotFoundError:
        return
